lib/opal/cli_options.rb

```diff
@@ -98,7 +98,7 @@ def initialize
         options[:runner_options][:port] = port.to_i
       end
 
-      on('-E', '--no-exit', 'Do not append a Kernel#exit at the end of file') do |no_exit|
+      on('-E', '--no-exit', 'Do not append a Kernel#exit at the end of file') do
```

Comment:
is it a bug or `no_exit` wasn't used ?

Reply:
It wasn't used, Ruby allows passing any amount of arguments to procs:
``` ruby
yield_self { |a, b = 1, *c, d| [a, b, c, d] }
 => [main, 1, [], nil]
```
I think it gets populated by the CLI options parser, so if you call `opal -E 1` this value becomes `"1"`. But this options acts as a flag, it doesn't matter which value is there.